File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/client/bind/ClientBindController.kt
package io.micronaut.docs.http.client.bind

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Header
import io.micronaut.http.annotation.QueryValue
import javax.annotation.Nullable

@Controller
class ClientBindController {

    @Get("/client/bind")
    fun test(@Header("X-Metadata-Version") version: String): String {
        return version
    }

    @Get("/client/authorized-resource{?name}")
    fun authorized(@QueryValue @Nullable name: String): String {
        return "Hello, $name"
    }
}

File: test-suite-python/src/test/python/micronaut/docs/http/client/bind/ClientBindController.py
from typing import Annotated

from jakarta.annotation import Nullable
from micronaut.http.annotation import Controller, Get, Header, QueryValue


@Controller
class ClientBindController:

    @Get("/client/bind")
    def test(self, version: Annotated[str, Header("X-Metadata-Version")]) -> str:
        return version

    @Get("/client/authorized-resource{?name}")
    def authorized(self, name: Annotated[str | None, QueryValue, Nullable]) -> str:
        return f"Hello, {name}"
